libcontainer/cgroups/systemd/v2.go

```diff
@@ -168,7 +174,19 @@ func (m *unifiedManager) Destroy() error {
 	if err != nil {
 		return err
 	}
-	dbusConnection.StopUnit(getUnitName(m.cgroups), "replace", nil)
+	unitName := getUnitName(m.cgroups)
+	statusChan := make(chan string, 1)
+	if _, err := dbusConnection.StopUnit(unitName, "replace", statusChan); err == nil {
+		select {
+		case s := <-statusChan:
+			// Please refer to https://godoc.org/github.com/coreos/go-systemd/dbus#Conn.StartUnit
+			if s != "done" {
+				logrus.Warnf("error removing unit `%s`: got `%s`. Continuing...", unitName, s)
+			}
```

Comment:
The channel can be closed here?

Reply:
I think we should not close the channel, because if someone send string to this closed channel, it will cause goroutine panic. Although I guess there is nobody would send sth to this channel.

Reply:
I checked the `github.com/coreos/go-systemd/dbus` code.

Looks like once it sends a string over the channel it forgets it. So yes we should close the channel -- unless we hit the timeout!

Reply:
finished now